From cb8188cdf6f8bbe139ff70b3e6203c5fe9c02187 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:06:09 -0400
Subject: [ethernet] qlcnic: Cleanup qlcnic_enable_msix() return values

Message-id: <1408626422-3941-86-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88616
O-Subject: [PATCH RHEL 7.1 085/138] qlcnic: Cleanup qlcnic_enable_msix() return values
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Alexander Gordeev <agordeev@redhat.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id 74a1fa45eba3958cbc6a6f726525e38263ea4cbf

Signed-off-by: Alexander Gordeev <agordeev@redhat.com>
Cc: Himanshu Madhani <himanshu.madhani@qlogic.com>
Cc: Rajesh Borundia <rajesh.borundia@qlogic.com>
Cc: Shahed Shaikh <shahed.shaikh@qlogic.com>
Cc: linux-driver@qlogic.com
Cc: netdev@vger.kernel.org
Cc: linux-pci@vger.kernel.org
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
index d355172..a436d0d 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_main.c
@@ -685,7 +685,7 @@ restore:
 int qlcnic_enable_msix(struct qlcnic_adapter *adapter, u32 num_msix)
 {
  struct pci_dev *pdev = adapter->pdev;
- int err = -1, vector;
+ int err, vector;
 
  if (!adapter->msix_entries) {
   adapter->msix_entries = kcalloc(num_msix,
@@ -707,7 +707,7 @@ enable_msix:
    adapter->flags |= QLCNIC_MSIX_ENABLED;
    adapter->ahw->num_msix = num_msix;
    dev_info(&pdev->dev, "using msi-x interrupts\n");
-   return err;
+   return 0;
   } else if (err > 0) {
    dev_info(&pdev->dev,
      "Unable to allocate %d MSI-X vectors, Available vectors %d\n",
@@ -716,12 +716,12 @@ enable_msix:
    if (qlcnic_82xx_check(adapter)) {
     num_msix = rounddown_pow_of_two(err);
     if (err < QLCNIC_82XX_MINIMUM_VECTOR)
-     return -EIO;
+     return -ENOSPC;
    } else {
     num_msix = rounddown_pow_of_two(err - 1);
     num_msix += 1;
     if (err < QLCNIC_83XX_MINIMUM_VECTOR)
-     return -EIO;
+     return -ENOSPC;
    }
 
    if (qlcnic_82xx_check(adapter) &&
@@ -748,7 +748,7 @@ enable_msix:
   }
  }
 
- return err;
+ return -EIO;
 }
 
 static int qlcnic_82xx_calculate_msix_vector(struct qlcnic_adapter *adapter)
-- 
1.7.1